doc/nrf/security/ap_protect.rst

```diff
@@ -170,43 +170,144 @@ Based on the available implementation types, you can configure the access port p
+TF-M is available for the nRF53 and nRF91 Series devices.
+
+While AP-Protect blocks access to all CPU registers and memories, Secure AP-Protect limits the CPU access to the non-secure side only.
+This allows debugging of the NSPE, while the SPE debugging is blocked.
+
+The following Kconfig options for enabling Secure AP-Protect are available for the nRF91x1 and nRF53 Series devices:
+
+* :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+* :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING`
+* :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USE_UICR`
+
+In addition, you can enable hardware Secure AP-Protect by setting the ``UICR.SECUREAPPROTECT`` register.
+
+Enabling software Secure AP-Protect with :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+-------------------------------------------------------------------------------------------
 
-To enable only the Secure AP-Protect, run the following command:
+Setting the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware enables the secure access protection mechanism for SoCs of the nRF53 Series.
+
+Enabling this Kconfig option writes the secure debugger register in the ``SystemInit()`` function to lock the secure access port protection at every boot.
+In addition to this, the ``UICR.SECUREAPPROTECT`` register should be written as instructed in :ref:`app_secure_approtect_uicr_approtect`.
+
+.. note::
+    For multi-image builds, this Kconfig option needs to be set for the first image (usually a bootloader).
+    Otherwise, the software Secure AP-Protect will not be sufficient as the debugger can be attached to the SPE after the first image opens the software Secure AP-Protect with the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USE_UICR` Kconfig option, which is the default value.
+
+    When using sysbuild, set the sysbuild Kconfig option ``SB_CONFIG_SECURE_APPROTECT_LOCK``, which enables the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option for all images.
+
+.. important::
+    On the nRF91x1 Series devices, the register setting related to the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option does not persist in System ON IDLE mode.
+    You must lock the ``UICR.SECUREAPPROTECT`` register to enable the hardware Secure AP-Protect mechanism as instructed in :ref:`app_secure_approtect_uicr_approtect`.
+
+Enabling software Secure AP-Protect with :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING`
+----------------------------------------------------------------------------------------------------
+
+Setting the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USE_UICR` Kconfig option to ``y`` disables the software Secure AP-Protect mechanism.
+This option does not touch the mechanism and keeps it closed.
```

Comment:
Closed how? What has closed it in that scenario?

Reply:
The AP-Protect and Secure AP-Protect start in a state where debugging is disabled, but can be later enabled. The user handling options leave it untouched.

Reply:
It is a bit complex.

By default, debugging is not disabled by hardware AP-Protect (UICR.APPROTECT).

Also by default, debugging is disabled by APPROTECT.DISABLE register (register name varies by device), which is then opened by (first image) in SystemInit() by writing open to it, if CONFIG_NRF_SECURE_APPROTECT_USE_UICR (default Kconfig) is set and UICR.APPROTECT is not set. 
If hardware AP-Protect is set by writing the UICR.APPROTECT, we also do not open the APPROTECT.DISABLE, but it does not really matter as hardware AP-Protect prevents debugger attachment.

Then we also have APPROTECT.LOCK register (register name varies by device), which is set by CONFIG_NRF_SECURE_APPROTECT_LOCK Kconfig, which prevents the APPROTECT.DISABLE from being opened, until the device is booted.

Then we have Secure versions of these registers for some products.